provider/maas/environ.go

```diff
@@ -2176,7 +2176,27 @@ func (env *maasEnviron) allocateContainerAddresses2(hostInstanceID instance.Id,
 	subnetToStaticRoutes := make(map[string][]gomaasapi.StaticRoute)
 	staticRoutes, err := env.maasController.StaticRoutes()
 	if err != nil {
-		return nil, errors.Annotate(err, "unable to look up static-routes")
+		// maas 2.0 does not support static-routes, so we just treat it as not
+		// having any static routes to worry about.
+		// Ideally we would know from the API version, or from the capabilities
```

Comment:
seems like TMI to me, can we just say:

> MAAS 2.0 does not support static-routes, and will return 404. MAAS
> does not report support for static-routes in its capabilities, so we treat
> 404 as meaning that it is unsupported and that there are no static routes
> to configure.

Other info could go in the commit message.